### PR TITLE
Add E2E for CAPV using vcsim

### PR DESCRIPTION
**Pull request by fabriziopandini**
/hold
for https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2767 to merge

cc @chrischdi

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/templates/cluster-api-provider-vsphere-presubmits.yaml.tpl [hunk 1]
```diff
@@ -201,7 +201,45 @@ presubmits:
...
       description: Runs all e2e tests
-{{ if eq $.branch "release-1.5" "release-1.6" "release-1.7" "release-1.8" | not }}
+{{ if eq $.branch "main" }}
+  - name: pull-cluster-api-provider-vsphere-e2e-vcsim-govmomi-{{ ReplaceAll $.branch "." "-" }}
+    branches:
+    - ^{{ $.branch }}$
+    labels:
+      preset-dind-enabled: "true"
+      preset-cluster-api-provider-vsphere-e2e-config: "true"
+      preset-cluster-api-provider-vsphere-gcs-creds: "true"
```
Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/templates/cluster-api-provider-vsphere-presubmits.yaml.tpl:211]: We should not need them here :-)

Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/templates/cluster-api-provider-vsphere-presubmits.yaml.tpl:211]:
```suggestion
```

**Review by chrischdi — COMMENTED** [review]
/lgtm
/approve

**Comment by fabriziopandini** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2767 merged
/hold cancel

**Pull request merged**